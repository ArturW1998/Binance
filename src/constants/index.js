export const ACTION_TYPES = {
  UPDATE_MARKET_PAIRS: 'UPDATE_MARKET_PAIRS',
  SET_ACTIVE_MARKET: 'SET_ACTIVE_MARKET',
  TOGGLE_SOCKET_STREAMS: 'TOGGLE_SOCKET_STREAMS',
};

export const PATHS = {
  BASE_PATH: 'wss://stream.binance.com',
  STREAM_PATH: '/stream',
  STREAM_PARAM: 'streams=',
};

export const LISTS = {
  PAIRS_LIST: ['BNB', 'BTC', 'ETH', 'USDT'],
  DOTS_LIST: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
  TABLE_DATA_LIST: [
    { title: 'Pair' },
    { title: 'Latest Price' },
    { title: 'Open' },
    { title: 'High' },
    { title: 'Low' },
    { title: 'Volume' },
  ],
};

export const CONFIG = {
  BASE_NAME: '/Binance/',
  LOCALSTORAGE_KEY: 'markets',
};
