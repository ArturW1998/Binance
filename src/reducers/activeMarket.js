import { SET_ACTIVE_MARKET } from '../constants';

export const initState = {};

const activeMarket = (state = initState, { type, data }) => {
  switch (type) {
    case SET_ACTIVE_MARKET:
      return { ...data };
    default:
      return state;
  }
};

export default activeMarket;
