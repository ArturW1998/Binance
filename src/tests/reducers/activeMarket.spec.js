import { SET_ACTIVE_MARKET } from '../../constants';
import activeMarket, { initState } from '../../reducers/activeMarket';

describe('activeMarket reducer', () => {
  const action = {
    type: SET_ACTIVE_MARKET,
    data: {
      market: 'BTC',
    },
  };

  const { data } = action;

  it(`${SET_ACTIVE_MARKET}`, () => {
    expect(activeMarket(initState, action)).toEqual(data);
  });

  it(`should handle ${SET_ACTIVE_MARKET}`, () => {
    expect(activeMarket(initState, action)).toMatchSnapshot();
  });
});

describe('activeMarket reducer initial state', () => {
  it('should return the initial state', () => {
    expect(activeMarket(initState, {})).toEqual(initState);
  });

  it('returns properly', () => {
    expect(activeMarket(initState, {})).toMatchSnapshot();
  });
});
