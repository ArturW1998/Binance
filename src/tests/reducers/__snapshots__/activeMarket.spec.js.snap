// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activeMarket reducer initial state returns properly 1`] = `Object {}`;

exports[`activeMarket reducer should handle SET_ACTIVE_MARKET 1`] = `
Object {
  "market": "BTC",
}
`;
